frontends/relay-futil/relay2futil_utilities.py

```diff
@@ -0,0 +1,149 @@
+import math
+from dataclasses import dataclass
+
+IdDictionary = {'cond': 0, 'std_const': 0, 'control': 0, 'group': 0, 'incr': 0, 'index': 0, 'incr': 0, 'let': 0,
+                'seq': 0, 'std_add': 0, 'std_eq': 0, 'std_le': 0, 'std_mem_d1': 0, 'std_mem_d2': 0, 'std_mem_d3': 0,
+                'std_reg': 0}
```

Comment:
Not necessary for this version of the PR (i.e., this can be addressed later), but it might be good to avoid mutable global state by making this an attribute on the `Relay2Futil` visitor class rather than a module-scope global. Then the dictionary would be an argument to the `id` function (which would still mutably update the dict).

Also, [defaultdict](https://docs.python.org/3.8/library/collections.html#collections.defaultdict) could be what the doctor ordered to avoid the need to list all the component names ahead of time.

Reply:
Agree 100% about the IdDictionary being in Relay2FuTIL. That's originally where ids were as well, I just moved it out of there to reason about things. I will fix this in the next CL. Thanks for looking!

And perfect, I was actually looking for that.